### PR TITLE
Migrate to pip-tools

### PR DESCRIPTION
**Pull request by offbyone**
I've been really liking the way pip-tools handles dependencies, as well as trying to get consistent code style enforced via [pre-commit](https://pre-commit.com/).

A side effect of running this through travis was learning that Python 3.5 isn't supported any more :(

### CODE REVIEW

**Comment by offbyone** [comment]
Also note that flake8 currently fails against some of the source, and that's okay; I want it in the pre-commit so that it slowly ratchets up quality as we touch files, not in one big gulp.

**Comment by brunns** [comment]
Nice. Do we still use tox for local development and stuff? Would tyhis add any steps to the relase process?

**Comment by brunns** [comment]
When you say Python 3.5 isn't supported any more - what is is not supported by? It's still working on Travis AFAIK. Not that I object to dropping it.

File: setup.py [hunk 1]
```diff
@@ -52,7 +58,6 @@ def read(fname):
         "Natural Language :: English",
         "Operating System :: OS Independent",
         "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3.5",
```
Comment (brunns) [inline setup.py:55]: Do we need to change `python_requires` too?

**Comment by brunns** [comment]
Shall I go ahead and merge this, then drop 3.5 support as another PR?

**Comment by offbyone** [comment]
Yeah, let's do that.

**Pull request merged**